### PR TITLE
Breaks the dependency on `log`

### PR DESCRIPTION
**Pull request by nblumhardt**
Part of #29.

I'm keen to rename `eminfo` -> `info` etc. as part of this ticket - thoughts?

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -91,9 +85,42 @@ pub mod formatters;
...
+    Info, 
+    Debug, 
+    Trace
+}
+
+static LOG_LEVEL_NAMES: [&'static str; 6] = ["OFF", "ERROR", "WARN", "INFO", "DEBUG", "TRACE"];
+
+impl fmt::Display for LogLevel {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        write!(fmt, "{}", LOG_LEVEL_NAMES[*self as usize])
```
Comment (KodrAus) [inline src/lib.rs]: That's a neat way of stringifying an enum... I wouldn't have thought of doing that

**Comment by KodrAus** [comment]
Yeh I agree that renaming the macros is a good idea. It'll make them feel more natural.

File: src/lib.rs (hunk 1 above)
Comment (nblumhardt) [inline src/lib.rs]: Mirrors what `std::log` does, not my innovation :-)

**Comment by nblumhardt** [comment]
Cool, I'll make those changes shortly.

**Comment by nblumhardt** [comment]
This is ready but for the unexpected build break on nightly. Investigating now...

**Comment by nblumhardt** [comment]
Ready to go! Downstream crates will need rebuilding once this is published, which I guess I'll do by hand after the merge.

**Comment by KodrAus** [comment]
I'll get `emit_elasticsearch` up to date

**Pull request merged**